### ISSUE TITLE
docker-compose status returns unhealthy while endpoint returns healthy

### ISSUE THREAD

**Issue by kamikazechaser** [milesmcc/shynet#183]
`docker-compose ps` returns:

```
      Name                    Command                   State                                        Ports                                 
-------------------------------------------------------------------------------------------------------------------------------------------
apps_shynet-db_1   docker-entrypoint.sh postgres    Up               5432/tcp                                                              
apps_shynet_1      ./entrypoint.sh                  Up (unhealthy)   0.0.0.0:49654->8080/tcp,:::49654->8080/tcp   
```

`curl 172.25.0.3:8080/healthz/?format=json` (Where the IP points to the container) returns:

```
{"Cache backend: default": "working", "DatabaseBackend": "working"}
```

Looks like unintended behaviour.

This prevents strict ingress controllers like Traefik from functioning because it would filter out anything that fails the health-check.

**Comment by milesmcc** [comment]
What is the HTTP status code of `curl 172.25.0.3:8080/healthz/?format=json`? What criteria does Traefik use to determine whether a health check is healthy or not?

**Comment by fnwbr** [comment]
I think I was seeing the same issue... this happens when `ALLOWED_HOSTS` is either unset or still set to the default `*`:

```
Performing startup checks...
Database is ready to go.
Startup checks complete!
Launching Shynet web server...
[2022-01-01 13:21:50 +0000] [1] [INFO] Starting gunicorn 20.1.0
[2022-01-01 13:21:50 +0000] [1] [INFO] Listening at: http://0.0.0.0:8080 (1)
[2022-01-01 13:21:50 +0000] [1] [INFO] Using worker: sync
[2022-01-01 13:21:50 +0000] [10] [INFO] Booting worker with pid: 10
ERROR Invalid HTTP_HOST header: '*'. The domain name provided is not valid according to RFC 1034/1035.
```

So I'm assuming the new health check is not yet working for that specific case of `ALLOWED_HOSTS` being set to `*`. After setting `ALLOWED_HOSTS` to a set of host names the health check was working and my container was starting as expected again.

**Comment by JuniorJPDJ** [comment]
Ugh, when designing healthcheck command I didn't consider allowed_hosts == "*"
I'll do pull request in a moment. Thanks for the issue.

**Comment by milesmcc** [comment]
> Ugh, when designing healthcheck command I didn't consider allowed_hosts == "*"
> I'll do pull request in a moment. Thanks for the issue.

Thanks for making this fix!

**Comment by JuniorJPDJ** [comment]
@fnwbr could you please check if this PR fixes your problem?

**Comment by fnwbr** [comment]
Hey, thanks for the quick fix! I'm unable to quicky test this out at the moment, sorry 😕 
Maybe @kamikazechaser can?

**Comment by kamikazechaser** [comment]
I'll try and test out the fix within the week and get back.

**Comment by kamikazechaser** [comment]
I tested the fix in #186 and it did not work. I have updated my original comment with the full docker-compose.yml that I am using with traefik.

**Comment by JuniorJPDJ** [comment]
Hmmm...

**Comment by JuniorJPDJ** [comment]
@kamikazechaser so you are not setting `ALLOWED_HOSTS` at all?

**Comment by sergioisidoro** [comment]
Also having this issue. Docker swarm was considering service unhealthy (ie "Starting"), and shutting it down when it timed out.

Turns out that for me the problem is that $PORT is not defined because the command not having a default, like many other places where PORT is used:
```
HEALTHCHECK CMD bash -c 'wget -o /dev/null -O /dev/null --header "Host: ${ALLOWED_HOSTS%%,*}" "http://127.0.0.1:$PORT/healthz/?format=json"'
```

After passing PORT=8080 in env it worked ok.

I'm not a super big fan of specifying the Health check in the Dockerfile, since you can do it in docker compose / docker swarm (leaving the choice to the user), but if envs are used, they should have a fallback.

**Comment by sergioisidoro** [comment]
I drafted something here: https://github.com/milesmcc/shynet/pull/249 I'm testing it now

**Comment by milesmcc** [comment]
@sergioisidoro it's been awhile since I explicitly published a new release. I just released v0.13.0 so that you can have a stable Docker image. (Everything was published on `latest`, but I understand that folks may not want to use that image.)

**Comment by djerfy** [comment]
> @sergioisidoro it's been awhile since I explicitly published a new release. I just released v0.13.0 so that you can have a stable Docker image. (Everything was published on `latest`, but I understand that folks may not want to use that image.)

Hi @milesmcc, sure? last gh-actions return an error during build: https://github.com/milesmcc/shynet/actions/runs/5577941089/jobs/10191428442

**Comment by milesmcc** [comment]
Yeah, it failed. Looks like the PyYAML dependency failed. I'm working right now, but if you/anyone else is able to debug the build and submit a PR, I'll merge + make a new release.

**Comment by djerfy** [comment]
> Yeah, it failed. Looks like the PyYAML dependency failed. I'm working right now, but if you/anyone else is able to debug the build and submit a PR, I'll merge + make a new release.

Yeah, PyYAML is impacted but caused by Cypthon 3.0 (released today). I found that for resolving:
```shell
poetry run pip install "Cython<3.0" "pyyaml==5.4.1" --no-build-isolation
```

Not better but works:
```shell
Installing dependencies from lock file
No dependencies to install or update
Installing the current project: shynet (0.13.0)
```
